src/middleware/authentication_middleware.ts

```diff
@@ -11,5 +11,7 @@ export const authenticationMiddleware = (req: Request, res: Response, next: Next
         returnUrl: BASE_URL + CHECK_SAVED_APPLICATION
     };
 
-    return acspProfileCreateAuthMiddleware(authMiddlewareConfig)(req, res, next);
+    return FEATURE_FLAG_VERIFY_SOLE_TRADER_ONLY === "true"
```

Comment:
functionality need to be reversed, as mentioned in the other comment, if the flag is true then verify sole trader only and not the other way round